src/main/resources/com/google/api/codegen/ruby/method_sample.snip

```diff
@@ -0,0 +1,134 @@
+@extends "ruby/common.snip"
+
+@snippet sampleCode(apiMethod)
+    @@example
+    @let lines = util.getDocLines(sampleCode_(apiMethod))
+        @join line : lines
+            {@""}  {@line}
+        @end
+    @end
+@end
+
+@snippet sampleCode_(apiMethod)
+    require "{@apiMethod.initCode.apiFileName}"
+
+    {@apiMethod.apiClassName} = {@apiMethod.initCode.packageName}\
+        ::{@apiMethod.apiClassName}
+    @join aliasDef : context.entrySet(apiMethod.initCode.typeTable.getImportsMap)
+        {@aliasDef.getValue} = {@aliasDef.getKey}
+    @end
+
+    {@apiMethod.apiVariableName} = {@apiMethod.apiClassName}.new
+    @if apiMethod.initCode.lines
+        {@initCode(apiMethod.initCode)}
+    @end
+    @switch apiMethod.type.toString
+    @case "FlattenedMethod"
+        {@flattenedMethodSampleCode(apiMethod)}
+    @case "PagedFlattenedMethod"
+        {@pagedFlattenedMethodSampleCode(apiMethod)}
+    @default
+        $unhandledCase: {@apiMethod.type.toString}$
+    @end
+@end
+
+@private flattenedMethodSampleCode(apiMethod)
+    @if apiMethod.hasReturnValue
+        response = {@methodCallSampleCode(apiMethod)}
+    @else
+        {@methodCallSampleCode(apiMethod)}
+    @end
+@end
+
+@private pagedFlattenedMethodSampleCode(apiMethod)
+
+    @# Iterate over all results.
+    {@methodCallSampleCode(apiMethod)}.each do |element|
+        @# Process element.
+    end
+
+    @# Or iterate over results one page at a time.
+    {@methodCallSampleCode(apiMethod)}.each_page do |page|
+      @# Process each page at a time.
+      page.each do |element|
+        @# Process element.
+      end
+    end
+@end
+
+@private methodCallSampleCode(apiMethod)
+    {@apiMethod.apiVariableName}.{@apiMethod.name}\
+        {@sampleMethodCallArgList(apiMethod.initCode.fieldSettings)}
+@end
+
+@private sampleMethodCallArgList(fieldSettings)
+    @if fieldSettings
```

Comment:
I'd prefer this `@if` to be performed in methodCallSampleCode. I think the parentheses are more an attribute of that logic than the sampleMethodCallArgList logic.